Python/aot_ceval_jit.c

```diff
@@ -2311,12 +2383,18 @@ static int emit_special_store_subscr(Jit* Dst, int inst_idx, int opcode, int opa
 
 // returns 0 if generation succeeded
 static int emit_special_compare_op(Jit* Dst, int oparg, RefStatus ref_status[2]) {
+#if PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION <= 8
     if (oparg != PyCmp_IS && oparg != PyCmp_IS_NOT) {
         return -1;
     }
+    int is_cmp = oparg == PyCmp_IS;
+#else
+    // for 3.9 we only call this function from IS_OP which means this but be a IS or IS_NOT comparison.
+    int is_cmp = oparg == 0;
```

Comment:
Sorry, does `oparg == 0` signify something?

Reply:
Yes they use it for `IS_OP` and `CONTAINS_OP` to signify that it's a `is not` / `not in` comparison:
```
int res = (left == right)^oparg;
PyObject *b = res ? Py_True : Py_False;
```
I will add a comment to explain it.

Reply:
Ohh I see -- the name `is_cmp`, especially with the alternate definition `oparg == PyCmp_IS`, read to me as "is this an `is` comparison", as opposed to dealing with negation